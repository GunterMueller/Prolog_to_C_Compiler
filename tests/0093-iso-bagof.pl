legs(A, 6) :- insect(A).
legs(A, 4) :- animal(A).
legs(A, 8) :- spider(A).

insect(bee).
insect(ant).
animal(horse).
animal(cat).
animal(dog).
spider(tarantula).

main :-
	findall(N/B, bagof(A, legs(A, N), B), L1),
	write(L1), nl,
	findall(S/Y, bagof(X, (X = Y; X = Z; Y = 1), S), L2),
	write(L2), nl.
