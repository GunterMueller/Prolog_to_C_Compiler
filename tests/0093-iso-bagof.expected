[4/[dog,cat,horse],6/[ant,bee],8/[tarantula]]
[[_766,_767]/_767,[_778]/1]
